firedrake/mesh.py

```diff
@@ -1787,50 +1793,21 @@ def input_ordering_sf(self):
         """
         if not isinstance(self.topology, VertexOnlyMeshTopology):
             raise AttributeError("Input ordering is only defined for vertex-only meshes.")
-        sf = PETSc.SF().create(comm=self.comm)
         nroots = self.input_ordering.num_cells()
-        input_ranks = self.topology_dm.getField("inputrank")
-        self.topology_dm.restoreField("inputrank")
-        input_indices = self.topology_dm.getField("inputindex")
-        self.topology_dm.restoreField("inputindex")
-        nleaves = len(input_ranks)
-        input_ranks_and_idxs = np.empty(2 * nleaves, dtype=IntType)
-        input_ranks_and_idxs[0::2] = input_ranks
-        input_ranks_and_idxs[1::2] = input_indices
-        # local looks like the below, which means we can just pass in None
-        # local = numpy.arange(nleaves, dtype=IntType)
-        sf.setGraph(nroots, None, input_ranks_and_idxs)
-        return sf
+        e_p_map = self.cell_closure[:, -1]
+        ilocal = np.empty_like(e_p_map)
+        if len(e_p_map) > 0:
+            cStart = e_p_map.min()
+            ilocal[e_p_map - cStart] = np.arange(len(e_p_map))
```

Comment:
How does this work? And aren't you remaking the same SF you used to make the swarm pointSF?

Reply:
`e_p_map` maps Firedrake cell numbers to swarm point numbers. `ilocal` is the inverse of this map, used in `_make_input_ordering_sf`.
The SF constructed here is different from the SF used to make pointSF: leaves of the former are Firedrake cell numbers and those of the latter are swarm point numbers.

Reply:
I think some comments and/or obvious variable naming are needed here then. For example, is `cStart` a cell number or an index? What do `e` and `p` stand for in `e_p_map`?

Reply:
Added.